app/utilities/drawing/master.py

```diff
@@ -0,0 +1,161 @@
+import sys
+import os
+import PIL
+from PIL import Image, ImageDraw, ImageOps, ImageFont, ImageEnhance, ImageFilter
+from pathlib import Path
+import requests
+from io import BytesIO
+
+from .utils import *
+from . import constants
+
+# changing import path
+abspath = os.path.abspath(__file__)
+ipath = os.path.dirname(abspath)
+os.chdir(ipath)
+
+
+def drawProfileCard(avatar_url, nickname, discriminator, labmem_number, level, rank, messages_sent, xp_current, next_level_xp, badges_list):
+    # getting base images
+    pfp_original = Image.open( requests.get(avatar_url, stream=True).raw )
+    background = PIL.Image.new(
+        mode="RGBA",
+        size=constants.background_dimensions,
+        color=(0, 0, 0, 255)
+    )
+
+    pfp_scaled = ImageOps.fit(pfp_original, constants.background_dimensions)
+    # putting scaled pfp on background
+    background = pfp_scaled.crop((
+        0, 0,
+        constants.background_dimensions[0],
+        constants.background_dimensions[1]
+    ))
+
+    # making background rounded rectangle (according to masks/background.png)
+    background = applyAlphaWithMask(background, 'masks/background.png')
+
+    # adding blurry region in the middle of background
+    cropped_img = background.crop((
+        constants.blurry_background_offset[0],
+        constants.blurry_background_offset[1],
+        constants.blurry_region_size[0]+constants.blurry_background_offset[0],
+        constants.blurry_region_size[1]+constants.blurry_background_offset[1]
+    ))
+
+    blurred_img = cropped_img.filter(ImageFilter.GaussianBlur(25)).convert("RGBA")
+    alpha_bg = PIL.Image.new(mode="RGBA", size=blurred_img.size, color=(18, 17, 21, 127))
+    blurred_img = Image.alpha_composite(blurred_img, alpha_bg)
+
+
+    background.paste(
+        blurred_img,
+        (constants.blurry_background_offset[0], constants.blurry_background_offset[1]),
+        create_rounded_rectangle_mask(blurred_img, constants.blurry_region_radius)
+    )
+
+    # adding semi-black-transparent rounded rectangle areas on top of blurried area
+    background = createSemiTransparentBlackRegionOnBackground(
+        constants.black_region_1_size,
+        constants.black_region_1_offset,
+        background
+    )
+    background = createSemiTransparentBlackRegionOnBackground(
+        constants.black_region_2_size,
+        constants.black_region_2_offset,
+        background
+    )
+
+    for i in range(0,7):
+        additional_x_offset = i * 10 + i * constants.badge_size[0]
+        badge_offset = (constants.first_badge_offset[0] + additional_x_offset, constants.first_badge_offset[1])
+        if len(badges_list) > i:
```

Comment:
We're making sure the current index is less than the number of badges, so this makes more sense.

```suggestion
        if i < len(badges_list):
```